zhaquirks/tuya/ts0601_din_power.py

```diff
@@ -177,6 +185,86 @@ def _update_attribute(self, attrid, value):
             self.endpoint.electrical_measurement.power_factor_reported(value / 10)
 
 
+class ZemismartManufCluster(TuyaManufClusterAttributes):
+    """Manufacturer Specific Cluster of the Zemismart SPM series Power Meter devices."""
+
+    attributes = {
+        ZEMISMART_TOTAL_ENERGY_ATTR: ("energy", t.uint32_t, True),
+        ZEMISMART_TOTAL_REVERSE_ENERGY_ATTR: ("reverse_energy", t.uint32_t, True),
+        ZEMISMART_VCP_ATTR: ("vcp_raw", t.data64, True),
+        ZEMISMART_VCP_P2_ATTR: ("vcp_p2_raw", t.data64, True),
+        ZEMISMART_VCP_P3_ATTR: ("vcp_p3_raw", t.data64, True),
+    }
+
+    def _update_attribute(self, attrid, value):
+        super()._update_attribute(attrid, value)
+        if attrid == ZEMISMART_TOTAL_ENERGY_ATTR:
+            self.endpoint.smartenergy_metering.energy_deliver_reported(value)
+        elif attrid == ZEMISMART_TOTAL_REVERSE_ENERGY_ATTR:
+            self.endpoint.smartenergy_metering.energy_receive_reported(value)
+        elif attrid == ZEMISMART_VCP_ATTR:
+            self.endpoint.electrical_measurement.vcp_reported(value, 0)
+        elif attrid == ZEMISMART_VCP_P2_ATTR:
+            self.endpoint.electrical_measurement.vcp_reported(value, 1)
+        elif attrid == ZEMISMART_VCP_P3_ATTR:
+            self.endpoint.electrical_measurement.vcp_reported(value, 2)
+
+
+class ZemismartPowerMeasurement(LocalDataCluster, ElectricalMeasurement):
+    """Custom class for power, voltage and current measurement."""
+
+    cluster_id = ElectricalMeasurement.cluster_id
```

Comment:
This line can still be removed, as `cluster_id` is set because of inheriting from `ElectricalMeasurement`.

(I think some of our docs need to be updated too)